test/system/sync.test.ts

```diff
@@ -165,13 +165,15 @@ describe('`snyk-api-import sync <...>`', () => {
         },
       },
       async (err, stdout, stderr) => {
-        expect(stderr).toEqual('');
+        expect(stderr).toEqual(
+          `Failed to sync target api-import-circle-test/deleted-repo. ERROR: Cannot read properties of undefined (reading 'branch')\n`,
```

Comment:
How come we need to change this?

Reply:
This test was actually broken but appeared to pass, because of the way the old Node version handled unhandled promise rejections.
It was expecting no `stderr` output, but was actually getting that error string, which triggered an `UnhandledPromiseRejectionWarning` and the test exited with code 0, so Jest saw it as a pass.
You can see an example of that on the master branch [here](https://app.circleci.com/pipelines/github/snyk/snyk-api-import/3549/workflows/2503dcc9-87ee-49fc-839f-2698b45c6a3f/jobs/3675).
<img width="2232" alt="Screenshot 2024-06-25 at 16 21 15" src="https://github.com/snyk/snyk-api-import/assets/10091929/9755268c-bc78-431e-a7af-7ab7e88660a0">

Reply:
It seems like a weird assertion for a test called "Successfully generates synced logs in --dryRun mode", especially leaking the typescript error regarding reading properties of undefined... I'm not sure we want to commit this change like this.

Reply:
It's not a TS error, it's actual output from the tool when attempting to read the branch of a target that has been deleted. The reason I left the assertion in is to ensure any additional stderr output will cause the test to fail, as it means something changed in the tool behaviour.

Reply:
Removed `stderr` assertion, in favour of adjusting error handling for that case - to be addressed in a future PR.